lib/galaxy/jobs/runners/util/cli/job/lsf.py

```diff
@@ -0,0 +1,110 @@
+# A simple CLI runner for slurm that can be used when running Galaxy from a
+# non-submit host and using a Slurm cluster.
+from logging import getLogger
+
+try:
+    from galaxy.model import Job
+    job_states = Job.states
+except ImportError:
+    # Not in Galaxy, map Galaxy job states to Pulsar ones.
+    from pulsar.util import enum
+    job_states = enum(RUNNING='running', OK='complete', QUEUED='queued', ERROR="failed")
+
+from ..job import BaseJobExec
+
+log = getLogger(__name__)
+
+argmap = {
+    'memory': '-M',  # There is code in job_script_kwargs relying on this name's setting
+    'cores': '-n',
+    'queue': '-q',
+    'working_dir': '-cwd'
+}
+
+
+class LSF(BaseJobExec):
+
+    def __init__(self, **params):
+        self.params = {}
+        for k, v in params.items():
+            self.params[k] = v
+
+    def job_script_kwargs(self, ofile, efile, job_name):
+        scriptargs = {'-o': ofile,
+                      '-e': efile,
+                      '-J': job_name}
+
+        # Map arguments using argmap.
+        for k, v in self.params.items():
+            if k == 'plugin':
+                continue
+            try:
+                if k == 'memory':
+                    # Memory requires both -m and -R rusage[mem=v] request
+                    scriptargs['-R'] = "\"rusage[mem=%s]\"" % v
+                if not k.startswith('-'):
+                    k = argmap[k]
+                scriptargs[k] = v
+            except Exception:
+                log.warning('Unrecognized long argument passed to LSF CLI plugin: %s' % k)
+
+        # Generated template.
+        template_scriptargs = ''
+        for k, v in scriptargs.items():
+            template_scriptargs += '#BSUB %s %s\n' % (k, v)
+        return dict(headers=template_scriptargs)
+
+    def submit(self, script_file):
+        # bsub returns Job <9147983> is submitted to default queue <research-rh7>.
+        # This should be really handled outside with something like
+        # parse_external. Currently CLI runner expect this to just send it in the last position
+        # of the string.
+        return "bsub <%s | awk '{ print $2}' | sed 's/[<>]//g'" % script_file
+
+    def delete(self, job_id):
+        return 'bkill %s' % job_id
+
+    def get_status(self, job_ids=None):
+        return "bjobs -a -o \"id stat\" -noheader"  # check this
+
+    def get_single_status(self, job_id):
+        return "bjobs -o stat -noheader " + job_id
+
+    def parse_status(self, status, job_ids):
+        # Get status for each job, skipping header.
+        rval = {}
+        for line in status.splitlines():
+            id, state = line.split()
```

Comment:
Some of the other runners can output something more structured, like xml or json, that might be an alternative if it exists ?  If not can you change `id` to `job_id` to not mask the builtin ?

Reply:
As you might see from lines 68 and 71 above I'm defining the output structure already... I would say that for one and two tokens that need to be generated in each case (only status for single case; job_id and status for the multiple case), this should be more than fine and I think is unnecessary to get any json/xml handling logic for such simple cases. Will definitely change id to job_id as suggested, thanks for spotting that.